next.clayui.com/content/docs/components/autocomplete.mdx

```diff
@@ -0,0 +1,134 @@
+---
+title: "Autocomplete"
+description: "An autocomplete text field is an input that offers the user text suggestions while they type."
+packageNpm: "@clayui/autocomplete"
+packageVersion: "3.0.0"
+packageStatus: "Beta"
+---
+
+import {AutocompleteWithLoadingState, AutocompleteWithDataProvider} from '../../../src/components/clay/Autocomplete';
+
+<div class="nav-toc">
+
+- [Item](#item)
+- [Loading state](#loading-state)
+- [Data from outside](#data-from-outside)
+- [API](#api)
+  - [Autocomplete](#autocomplete)
+  - [Autocomplete.DropDown](#autocomplete.dropdown)
+  - [Autocomplete.Input](#autocomplete.input)
+  - [Autocomplete.Item](#autocomplete.item)
+  - [Autocomplete.LoadingIndicator](#autocomplete.loadingindicator)
+
+</div>
+
+<div class="clay-site-alert alert alert-info">
+	Check the <a href="https://liferay.design/lexicon/">Lexicon</a> <a href="https://liferay.design/lexicon/core-components/forms/text-input-variations/">Autocomplete Pattern</a> for a more in-depth look at the motivations and proper usage of this component.
+</div>
+
+Autocomplete provides the necessary components for you to compose with the other components, **by itself it does not do much**, you need to use the `<ClayDropDown />` along with `<Autocomplete.DropDown />` and `<Autocomplete.Item />` to create the suggestions.
+
+```jsx
+<ClayAutocomplete.DropDown>
+	<ClayDropDown.ItemList>
```

Comment:
I'd rather keep separate, I think export like `<Autocomplete.ItemList />` takes away the beauty of seeing the composition between components and how they can relate. And adding it to `<ClayAutocomplete.DropDown />` can take away other customization possibilities. For example, I can add headers to my list. Probably I could create several `<ClayDropDown.ItemList />` inside the `<ClayAutocomplete.DropDown>`.

Reply:
The main concern I have is that `@clayui/dropdown` is now a peer dependency of autocomplete since we now always expect the user to use it when they render an autocomplete list. Really the user doesn't need to know that Autocomplete is the composition of using dropdown, the user just wants to use the component for its intended purpose.

Reply:
Yes we can agree, we will have people who just want to use the component without needing to know that it needs to compose things, probably these people are the great candidates to use the high-level components, probably those who want to create their own autocomplete with their rules, will likely compose with other components.

Reply:
Yeah makes sense, if we do keep this API by keeping dropdown separate, we will need a peerDependency of dropdown right?

Reply:
Maybe, I think we should change all the dependencies we have between the components for `peerDependency`, currently they are like` dependencies` the dropdown also in that case.